### PR TITLE
Add support for multiple different sizes per module

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
* Stats include `sizes` next to `size` for modules and chunks
* `Module.size` now takes an optional type argument
* `Generator` has two new methods `getSize` `getTypes`
* `Module.getSourceTypes()` now returns available source types
* `splitChunks.min/maxSize` can now be objects specifying different sizes. Default is `javascript` size.

### CODE REVIEW

File: lib/optimize/SplitChunksPlugin.js [hunk 1]
```diff
@@ -28,10 +28,11 @@ const MinMaxSizeWarning = require("./MinMaxSizeWarning");
 
 /**
  * @typedef {Object} ChunksInfoItem
- * @property {SortableSet<Module>} modules
+ * @property {SortableSet} modules
```
Comment (ooflorent) [inline lib/optimize/SplitChunksPlugin.js]: Why did you changed that?

File: lib/wasm/WebAssemblyJavascriptGenerator.js [hunk 2]
```diff
@@ -18,7 +18,25 @@ const WebAssemblyImportDependency = require("../dependencies/WebAssemblyImportDe
...
+		return TYPES;
+	}
+
+	/**
+	 * @param {NormalModule} module the module
+	 * @param {string=} type source type
+	 * @returns {number} estimate size of the module
+	 */
+	getSize(module, type) {
+		return 100 + module.dependencies.length * 5;
```
Comment (ooflorent) [inline lib/wasm/WebAssemblyJavascriptGenerator.js]: What's the reasoning behind this formula?

File: lib/optimize/SplitChunksPlugin.js (hunk 1 above)
Comment (sokra) [reply lib/optimize/SplitChunksPlugin.js]: I moved the duplicate typedef from below. I'll restore it.

File: lib/wasm/WebAssemblyJavascriptGenerator.js (hunk 2 above)
Comment (sokra) [reply lib/wasm/WebAssemblyJavascriptGenerator.js]: That's about the code size of the generated code for WebAssembly modules.

**Pull request merged**